flink/src/main/java/org/apache/iceberg/flink/sink/FlinkSink.java

```diff
@@ -412,10 +417,16 @@ private String operatorName(String suffix) {
 
       switch (writeMode) {
         case NONE:
+          if (!equalityFieldIds.isEmpty()) {
+            return input.keyBy(new EqualityFieldKeySelector(equalityFieldIds, iSchema, flinkRowType));
```

Comment:
As the `RANGE` distribution mode will also fallback to use `NONE` mode,  shouldn't we also use the same shuffle strategy for it ?

Reply:
See: https://github.com/apache/iceberg/pull/2898/files#diff-a26cb8b3f0f5b0e6a9f110bb0b3fdd0ed625bfc9faf27e248f6446818cad4d69R436

Reply:
Yes! `RANGE` distribution mode should use the same shuffle strategy,  I miss this case, thanks for your remind！

Reply:
I am wondering if we can have explicit `DistributionMode` to shuffle by equality columns. Current `HASH` mode is more like `HASH-BY-PARTITION-COLUMNS`. We can have a `HASH-BY-EQUALITY-COLUMNS`. We can discuss the names, but you should get my idea.